Plugins: Display chat info in checkout summary for plugins

epeicher opened the pull request:
The live chat support information was not displayed in the checkout
summary when purchasing individual plugins.

To display the live chat support, the current plan information is
extracted using selectors. The following criteria are required to allow
live chat support:
- Plan type
- ~Billing period~ Plan period

Also, the live chat support is already displayed as part of the plan
summary so it will only be displayed when not purchasing a plan.

#### Changes proposed in this Pull Request

* Include the `Live chat support` message in the checkout details card that will be displayed if:
  * There is no plan in cart
  * The user plan is either Premium, Business or eCommerce. 
    :warning: **Note:** Currently Premium plans cannot purchase plugins but the logic has been [copied from getPlanFeatures](https://github.com/Automattic/wp-calypso/blob/4ef9895f501b61e017f0b8b8edaacdb2a4088f5e/client/my-sites/checkout/composite-checkout/lib/get-plan-features.ts#L13)
  * The ~plugin~ plan is billed _Annually_
* The existing logic that displays features when a plan is purchased has not been modified

|Before | After|
|-------|------|
|![CleanShot 2022-02-22 at 09 38 24@2x](https://user-images.githubusercontent.com/3519124/155094341-917703ae-178f-4744-9c8e-4b2adb6e4e2d.png)|![CleanShot 2022-02-22 at 09 42 18@2x](https://user-images.githubusercontent.com/3519124/155094990-14fcc679-176b-4684-8d30-c0616eefcaaf.png)|

#### Testing instructions

##### Scenario 1
* Open a site with either a Business or eCommerce plan _which is **not** billed Monthly_
* Add a paid plugin ~and select to pay _Annually_~
* Click on Purchase and Activate
* If a prompt is displayed to notify that the site will be moved, click Continue
* The card with the title `Included with your purchase` on the top right should include `Live chat support`

##### Scenario 2
* Open a site with either a Business or eCommerce plan _which is billed Monthly_
* Add a paid plugin ~and select to pay _Monthly_~
* Click on Purchase and Activate
* If a prompt is displayed to notify that the site will be moved, click Continue
* The card with the title `Included with your purchase` on the top right should **not** include `Live chat support`

Fixes #61207

epeicher committed:
Plugins: Display chat info in checkout summary for plugins

The live chat support information was not displayed in the checkout
summary when purchasing individual plugins.

To display the live chat support, the current plan information is
extracted using selectors. The following criteria is required to allow
live chat support:
- Plan type
- Billing period

Also, the live chat support is already displayed as part of the plan
summary so it will only be displayed when not purchasing a plan.

epeicher committed:
Refactor checkout chat support to external component

Extracts the logic and display of the chat support in the
checkout details card to an external component. Leaves the
condition `hasPlanInCart` in parent component so it can be easily
identified that the new logic is applied when the plan features are
not applied.

epeicher committed:
Add slice of state to tests that selector does not fail

This is to mock the `getBillingInterval` selector

cpapazoglou commented on client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.tsx:
```diff
@@ -288,6 +297,34 @@ function CheckoutSummaryPlanFeatures( props: {
+	);
+	const translate = useTranslate();
+
+	const billingPeriod = useSelector( getBillingInterval );
+	const isAnnualPeriod = billingPeriod === IntervalLength.ANNUALLY;
+
+	const currentPlanSlug = currentPlan?.productSlug;
+
+	const isChatAvailable =
+		currentPlanSlug &&
+		( isWpComPremiumPlan( currentPlanSlug ) ||
+			isWpComBusinessPlan( currentPlanSlug ) ||
+			isWpComEcommercePlan( currentPlanSlug ) ) &&
+		isAnnualPeriod;
```
Live chat support is depending on the plan billing period not on the `getBillingInterval` which is what user has selected in the Plugins UI

![CleanShot 2022-02-22 at 11 31 00@2x](https://user-images.githubusercontent.com/12430020/155103391-e3c18a90-8c0f-479a-8e0f-8ed89956533a.png)

epeicher replied:
Thanks for your comments @cpapazoglou . 

I thought about that condition but the reason why I have used the billing interval _of the plugin_ as opposed to the _plan_ billing period is that when purchasing a plugin, the `Live chat support` is displayed depending on the _plugin_ billing interval 
|Monthly|Annual|
|--|--|
|![CleanShot 2022-02-22 at 10 43 28@2x](https://user-images.githubusercontent.com/3519124/155106230-95fbcfbe-7299-450b-83ca-f39772184a4e.png)|![CleanShot 2022-02-22 at 10 44 49@2x](https://user-images.githubusercontent.com/3519124/155106089-f7bd0da1-548d-45bf-a46d-af1ef5ff2251.png)|

So it seems to be inconsistent with the change above, what do you think about it?

cpapazoglou replied:
That seems like a bug we need also to address, my mistake 🙈. 

We should depend on the plugin billing interval only when user needs to also upgrade their plan and thus we add a plan to their cart. On the rest of the cases, we should depend on the plan interval.

epeicher replied:
No worries, I should have clarified the change in the Plugin details page.

I will update the code and the PR to address your comments and create an issue for the Plugin details page message described in the above screenshots.

epeicher replied:
Bug #61365 created

cpapazoglou reviewed (changes requested):
Thanks for the detailed PR description @epeicher.

While it works as you described, the "Live Chat" depends only in the plan (type and billing interval). The following should be the condition, after retrieving the billing interval of the plan.

`(Premium || Business || Ecommerce) && ( annual || biannual )`

epeicher committed:
Use plan period to allow live chat support

Instead of using the plugin period, the plan period is used to display or
hide live chat support.

This change also removes the slice of mock state added for tests that is
not required anymore after removing the selector that used it.

cpapazoglou reviewed (approved):
Followed the test scenarios and works as described.

Whatever you buy, if you got a non - monthly plan then you get Live chat support. Which is what we advertise

![CleanShot 2022-02-22 at 15 47 54@2x](https://user-images.githubusercontent.com/12430020/155145219-6f92889a-66a8-4bd1-ab20-bb6ca8e797ae.png)

cpapazoglou commented:
The failing e2e tests seem unrelated to the code changes, I think we are OK to deploy.
